client/src/pages/course/student/auto-test.tsx

```diff
@@ -349,16 +355,32 @@ function getAttemptsLeftMessage(value: number, strictAttemptsMode: boolean) {
   return 'Limit of "free" attempts is over. Now you can get only half a score.';
 }
 
+function getTimeToTheNextSubmit(hours: number, lastAttemptTime?: string) {
+  if (!hours || !lastAttemptTime) return 0;
+  const diff = new Date(lastAttemptTime).getTime() - new Date().getTime() + hours * 60 * 60 * 1000;
```

Comment:
since we already use moment here, maybe try to make it more readable?

```typescript
const diff = moment(lastAttemptTime).subtract(hours, 'hour').diff(moment())
```

Reply:
Fixed